src/lib/coda_numbers/nat.ml

```diff
@@ -46,6 +46,7 @@ struct
       Integer.constant ~length:N.length_in_bits ~m
         (Bignum_bigint.of_int (N.to_int n))
 
+    (* warning: this typ does not work correctly with the generic if_ *)
```

Comment:
Can you duplicate this as a doc-comment in `intf.ml`, so that Merlin will pick it up?